### PR TITLE
feat(core): introduce DaffNumericallyPaginable for collections

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

**Review by griest024 — APPROVED** [review]

File: libs/core/state/src/collection/paginateable.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface DaffNumericallyPaginateable {
```
Comment (lderrickable) [inline libs/core/state/src/collection/paginateable.ts:1]: paginable?

Comment (damienwebdev) [reply libs/core/state/src/collection/paginateable.ts:1]: I was torn here. I couldn't decide on the adjective form here. I googled for 

```txt
Paginatable 
Paginateable 
Paginable 
Pageable
```

Of these four, none appear in webster's.

Comment (griest024) [reply libs/core/state/src/collection/paginateable.ts:1]: I like Paginable as well.

Comment (damienwebdev) [reply libs/core/state/src/collection/paginateable.ts]: 6a032467

**Review by griest024 — APPROVED** [review]

**Pull request merged**